sktime/forecasting/adapters/_autots.py

```diff
@@ -0,0 +1,449 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+from __future__ import annotations
+
+__author__ = ["MBristle"]
+
+import logging
+
+import pandas as pd
+
+from sktime.forecasting.base import BaseForecaster, ForecastingHorizon
+
+
+class AutoTSAdapter(BaseForecaster):
+    """Act as adapter for the AutoTS library.
+
+    Parameter:
+    ---------
+        fh (ForecastingHorizon): The forecasting horizon.
+        model_list (str): The list of models to use. See docs for details.
```

Comment:
I'm aware this is a boatload of parameters - still it would be nice to have a much easier way to have a list of valid values or a quick link to it. The defaults should also be listed.

For that, and formatting (already mentioned by @yarnabrina), have a look at any other transformer in `sktime`.

Reply:
Yep, I agree. I just hope the interface is not too volatile, as maintenance will be a pain otherwise.

Reply:
a prime example of volatile interfaces is `statsforecast` - every version seems to add, remove, change defaults or names of parameters. The adapter class maintained by @yarnabrina - `_GeneralizedStatsForecastAdapter` - has some defensive mechanisms for that reason, which could be copied, or even factored into a "mother adapter class", should we need them.